client/errors.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+// Error is the real value of response.Result when an error type response occurs.
+type Error struct {
+	Message string      `json:"message"`
+	Kind    ErrorKind   `json:"kind,omitempty"`
+	Value   interface{} `json:"value,omitempty"`
+}
+
+func (e *Error) Error() string {
+	return e.Message
+}
+
+type ErrorKind string
+
+// Error kinds for use as a response result.
+const (
+	ErrorKindLoginRequired     = ErrorKind("login-required")
+	ErrorKindNoDefaultServices = ErrorKind("no-default-services")
+	ErrorKindNotFound          = ErrorKind("not-found")
+	ErrorKindPermissionDenied  = ErrorKind("permission-denied")
+	ErrorKindGenericFileError  = ErrorKind("generic-file-error")
```

Comment:
This might as well be `ErrorKindFileError`

and also

![8020mi](https://github.com/canonical/pebble/assets/1417596/c577d6bd-8706-4fe5-8c01-210b5a431d21)

Reply:
While `not-found` and `permission-denied` are typically used in contexts involving files, I think those are generic enough that can be used to indicate e.g. the non-existence of a service, so these shouldn't really be in a "file management errors" bag.

Also, there's a lot of stuff that can go wrong when dealing with files (e.g. `EBUSY`, `EXDEV`) for which we don't have suitable error kinds.

Reply:
As per Gustavo's clarification, we cannot change the error content at this stage.